## Task 13 — Upload Traffic by Country

### Business Question

Which countries have the highest average upload traffic per customer?

### Analysis

Group customers by country, calculate the number of customers and total upload traffic for each country, then calculate the average upload traffic per customer and sort the results in descending order.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
upload_traffic_by_country = starlink.groupby("country").agg(
    users_amount = ("unique_id","count"),
    total_gb_uploaded = ("gb_uploaded","sum")
)
upload_traffic_by_country["average_gb_uploaded_per_user"] = upload_traffic_by_country["total_gb_uploaded"] / upload_traffic_by_country["users_amount"]
upload_traffic_by_country_sorted = upload_traffic_by_country.sort_values(
    "average_gb_uploaded_per_user",
    ascending = False
)
print(upload_traffic_by_country_sorted)

                users_amount  total_gb_uploaded  average_gb_uploaded_per_user
country                                                                      
Ukraine                 1032          277490.98                    268.886609
United Kingdom           977          261898.77                    268.064248
Spain                   1019          268549.10                    263.541806
Germany                 1027          269388.56                    262.306290
United States            954          249946.39                    261.998312
Canada                   997          256590.04                    257.362126
Australia               1030          264678.26                    256.969184
France                   996          255298.49                    256.323785
Brazil                   996          253390.40                    254.408032
Japan                    973          245148.97                    251.951665


### Result

Ukraine has the highest average upload traffic per customer at 268.89 GB, followed by the United Kingdom at 268.06 GB and Spain at 263.54 GB. Japan has the lowest average upload traffic at 251.95 GB per customer.

### Business Insight

Average upload traffic varies moderately across countries, with Ukraine showing the highest value. The relatively small difference between the countries suggests that upload activity is fairly consistent across the analyzed markets.